wgengine/magicsock/magicsock.go

```diff
@@ -5105,6 +5125,88 @@ func (s derpAddrFamSelector) PreferIPv6() bool {
 	return false
 }
 
+type endpointTrackerEntry struct {
+	endpoint tailcfg.Endpoint
+	until    time.Time
+}
+
+type endpointTracker struct {
+	mu    sync.Mutex
+	cache map[netip.AddrPort]endpointTrackerEntry
+}
+
+const endpointTrackerLifetime = 5*time.Minute + 10*time.Second // slightly longer than a full netcheck period
+
+func (et *endpointTracker) update(now time.Time, eps []tailcfg.Endpoint) (ret []tailcfg.Endpoint) {
+	ret = eps
+
+	// We can have a very large set of endpoints for machines with many
```

Comment:
Making a couple dicts to diff and update a list is a fairly common pattern in Go, so IMO there's no need to spell it out. It made me double-take a bit and hunt for what might be different, when nothing was.

Reply:
Removed this comment now that we're using a `set.Slice` 👍